### PR TITLE
[Terra_2_NCBI] remove metadata requirements with skip_biosample == true

### LINKED ISSUE #435: [Terra_2_NCBI] skip-biosample should prevent biosample value checks
:bug:

### :pencil: Describe the Issue

A Terra user was trying to use it to submit reads to SRA for samples that already have biosample accessions, so uploaded the biosample accessions to terra and ran the workflow with skip_biosample = true. But it looks like because they didn't upload the required biosample metadata, the workflow failed because the prune_table task excluded the samples for lacking the metadata.

### PR DESCRIPTION
**Pull request by sage-wright**
This PR closes #435 

🗑️ This dev branch should <NOT> be deleted after merging to main.

## :brain: Aim, Context and Functionality

`skip_biosample` should avoid checking for biosample metadata, but it was not. :( Tragic. Anyways, now it skips that check so you don't need to provide it anymore.

## :hammer_and_wrench:  Impacted Workflows/Tasks & Changes Being Made
This will affect the behavior of the workflow(s) even if users don’t change any workflow inputs relative to the last version : Yes/No

Running this workflow on different occasions could result in different results, e.g. due to use of a live database, "latest" docker image, or stochastic data processing : Yes/No 

`biosample_type` is now an optional variable

#### 🔄 Data Processing 

Docker/software or software versions changed: 

Databases or database versions changed:

Data processing/commands changed:

Checking for biosample metadata is skipped if you skip biosample submission/creation

File processing changed:

Compute resources changed:

## :microscope: Final Developer Checklist

- [x] The workflow/task has been tested locally and results, including file contents, are as anticipated
- [ ] The workflow/task has been tested on Terra and results, including file contents, are as anticipated
- [x] The CI/CD has been adjusted and tests are passing (to be completed by Theiagen developer)
- [x] Code changes follow the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-bb456f34322d4f4db699d4029050481c)

## 🎯 Reviewer Checklist 

- [ ] All impacted workflows/tasks have been tested on Terra with a different dataset than used for development
- [x] All reviewer-suggested scenarios have been tested and any additional
- [ ] All changed results have been confirmed to be accurate
- [ ] All workflows/tasks impacted by change/s have been tested using a standard validation dataset to ensure no unintended change of functionality
- [x] All code adheres to the style guide
- [x] MD5 sums have been updated
- [ ] The PR author has addressed all comments

## 🗂️ Associated Documentation (to be completed by Theiagen developer)

- [ ] Relevant documentation on the Public Health Resources "PHB Main" has been updated
- [ ] Workflow diagrams have been updated to reflect changes

### CODE REVIEW

**Comment by kevinlibuit** [comment]
Notes: @sage-wright tested function of this change locally; @AndrewLangvt and I launching a job in Terra to verify https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Scribner_Sandbox/job_history/1d12ad20-d3d7-4279-8516-2aeaa9dd31bd

File: tasks/utilities/submission/task_submission.wdl [hunk 1]
```diff
@@ -60,62 +60,63 @@ task prune_table {
...
-      #   coverage after trimming >= 20X
-      #if "mean_coverage_depth" in table.columns:
-      #  table = table[(table.mean_coverage_depth > 20)]
-
-    elif ("~{biosample_type}".lower() == "virus"):
-      required_metadata = ["submission_id", "organism", "isolate", "collection_date", "geo_loc_name", "isolation_source"]
-      optional_metadata = ["sample_title", "bioprojection_accession","attribute_package", "host", "lab_host", "altitude", "biomaterial_provider", "collected_by", "culture_collection", "depth", "disease", "env_broad_scale", "genotype", "host_tissue_sampled", "identified_by", "lat_lon", "passage_history", "samp_size", "serotype", "specimen_voucher", "strain", "temp", "description"]
-
-      table["attribute_package"] = "Virus.1.0"
+    if (os.environ["skip_bio"] == "false"):
```
Comment (kevinlibuit) [inline tasks/utilities/submission/task_submission.wdl:63]: Seems to be an upstream issue with the `skip_bio` flag: in line 33 above, looks as if any value, even `"false"` would be converted to a `true` boolean value. @sage-wright am I reading this correctly?

**Comment by kevinlibuit** [comment]
Seeing job failures in Terra. Need to assess further: https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Scribner_Sandbox/job_history/1d12ad20-d3d7-4279-8516-2aeaa9dd31bd

File: tasks/utilities/submission/task_submission.wdl (hunk 1 above)
Comment (sage-wright) [reply tasks/utilities/submission/task_submission.wdl:63]: No, the `skip_bio` flag would not be converted to a true Boolean even if it was false

**Review by AndrewLangvt — APPROVED** [review]
Tested via terra with `skip_biosample` set to `true` and workflow succeeds https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Scribner_Sandbox/job_history/67b398c6-6ee9-48ce-b171-9836337b489c

Tested via Terra with `skip_biosample` boolean not specified. Process fails, as expected, due to lack of metadata in table 
https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Scribner_Sandbox/job_history/72d30477-59f2-4c57-9ab1-f21f7315e6f8

**Pull request merged**